src/methods/SANA.cpp

```diff
@@ -2273,6 +2297,41 @@ int SANA::MS3IncSwapOp(Job &job, uint source1, uint source2, uint target1, uint
     return 0;
 }
 int SANA::MS3IncSwapOp(uint source1, uint source2, uint target1, uint target2) {
+    if (MultiS3::_type==1){
```

Comment:
I don't think you need anything inside this "if" statement. The whole point of using edge exposure is that *it's already computed for you*, including the incremental evaluation; I think the variable you want is "exposedEdgesNumer".  So that's the value you use when _type==1. Get rid of this whole section.

Reply:
MS3IncSwapOp and MS3IncChangeOp are for calculating the numerator of MS3. For the ee variant, the numerator is the number of edges under the small network + the number of not-lonely edges in the small network. 
Because we haven't figured out why the default version uses total_degree for calculating ms3(which might be another variant of ms3), so, I wrote theses sections of the code to ensure the numerator of the ee variant is properly calculated.

Reply:
Aha.  My mistake. Thanks.